### ISSUE TITLE
Add stty applet

### ISSUE THREAD

**Issue by dgtlrift** [rmyorston/busybox-w32#58]
Add stty applet to allow manipulation of text interface column width, etc

**Comment by rmyorston** [comment]
I've added some infrastructure to make it possible to build ```stty``` (and the non-standard ```ttysize``` applet). This is in the stty branch.

Since Windows consoles are quite different from ttys most ```stty``` options don't work properly. It's possible to get the terminal size (```stty size```) and set the width (```stty cols NN```) or height (```stty rows NN```). At least, these work in Windows 8.1. In Windows 10, for some reason I've been unable to determine, changing the console size adjusts the size of the window but ```stty size``` reports the previous dimensions.

Also, the applet should probably check that the new console size is consistent with the console buffer size and adjust the buffer if necessary.

I see that the MKS Toolkit has a version of [```stty```](https://www.mkssoftware.com/docs/man1/stty.1.asp) which exposes more of the Windows console API (and fewer of the traditional options). Doing something like that would require considerable surgery to the applet: the current version doesn't change the applet at all, just some of the WIN32 compatibility code.

**Comment by rmyorston** [comment]
The problem with changing the console size seems to have been fixed in the Windows 10 Anniversary Update.

**Comment by dgtlrift** [comment]
I tried to build the stty branch but have run into some errors on the error with types in coreutils/stty.c - is this branch obsoleted by the console size fixes introduced in Win 10 Ann?

**Comment by rmyorston** [comment]
No, not obsolete.

What errors are you getting?   I don't see any problem building on Fedora 24 (gcc 6.1) or Fedora 23 (gcc 5.2).

**Comment by mentalisttraceur** [comment]
If I may propose a contrary view: Is this really something that busybox-w32 needs to handle? A part of me is inclined to say that the problem domain on Windows for `stty` is very different from Unix-like environments - maybe it's best left to an external tool that can account for the numerous Windows-specific differences and compensate for portability issues between different Windows versions and the different console emulators/wrappers that exist for Windows?

**Comment by stianhoiland** [comment]
I found myself wanting to implement some scripts that require `stty` for raw + noecho This should be implementable with some calls to `SetConsoleMode` (see [High-Level Console Modes](https://learn.microsoft.com/en-us/windows/console/high-level-console-modes)). With Microsoft Terminal, Microsoft is moving away from the old console API to virtual terminal sequences (see [Console Virtual Terminal Sequences](https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences) & [Classic Console APIs versus Virtual Terminal Sequences](https://learn.microsoft.com/en-us/windows/console/classic-vs-vt)).

**Comment by avih** [comment]
It's already using VT mode where possible (win10+), which is how 256/true color sequences work on win10+, while on earlier windows it's translated to 8/16 colors by the internal terminal emulation code in busybox-w32.

I think cursor-moving sequences and others already go directly to the terminal in VT mode, and I don't recall whether alt-screen (used for instance in `less` or `vi`) in VT mode is passthrough to the terminal - or still emulated in busybox-w32.

Regardless of an actual stty applet, my hunch is that adding support for various (VT) terminal modes would not be simple, because at the very least input is tied quite strongly to the Console API (and its issues and bugs) rather than reading raw data from the terminal file descriptor. I don't even know whether the latter is possible on Windows (it might be, but it's not obvious to me).

It definitely could be interesting, and open the door to classes of scripts which currently don't work in busybox-w32, I think mainly those which handle raw non-printable keys - like editing keys, or mouse wheel, etc.

However, additionally, where possible, busybox-w32 tries hard to work the same in pre-win10 via internal terminal emulation, and while some things are simply impossible - like more than 16 colors at the console window, raw input/output emulation should be technically possible, but it may require considerable amount of work to behave reasonably via emulation, and I don't know whether it's worth the effort, and/or the further diversion between win10+ and pre-10.

Bottom line, I think it would be nice, but it might not be simple or even reasonably possible, and potentially further increase the behavior gap between win10 and earlier versions. But if someone wants to have a go, it could be interesting to see how the code looks like.
